src/app/pages/private/collection/collection.page.ts

```diff
@@ -59,11 +78,43 @@ export default class CollectionPage {
   rarityTotalCardCount = signal<number>(20);
   rarities = signal<string[]>(stubRarityList);
 
+  cardsList = signal<Card[]>(stubCardList);
+  searchList = signal<Card[] | null>(null);
+  isSearchActive = signal<boolean>(false);
+
   constructor() {
     addIcons({ settingsOutline });
   }
 
+  ngAfterViewInit(): void {
```

Comment:
Try to create behaviour subject, that will send API request when received new search value.